### PR TITLE
Update according Truffle team

### PR DESCRIPTION
**Pull request by solangegueiros**
## What

- Add more explanations in readme
- Add links to RSK, tutorial rsk-react-express-box and gitter
- Update method for get gas price

## Why

- Truffle team approved other tutorial in this style
- To be more friendly
- To link to the tutorial in DevPortal

## Refs

- [Trello card] (https://trello.com/c/YYay3DI4)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,59 +1,207 @@
...
+
+4. Take a look at the smart contract `SimpleStorage.sol`. You can check it out in folder `contracts`.
+
+This smart contract has:
+
+* A variable `storedData` to store a number
+* A function `get()` to return the number stored at variable `storedData`
+* A function `set()` to change the number stored at variable `storedData`
+
+5. Compile and migrate the smart contract. Note inside the development console we don't preface commands with `truffle`.
```
Comment (owans) [inline README.md]:
```suggestion
5. Compile and migrate the smart contract. Note, inside the development console we don't preface commands with `truffle`.
```

File: README.md [hunk 2]
```diff
@@ -1,59 +1,207 @@
...
+migrate
+```
+
+And the `migrate output` should be similar to:
+
+![truffle migrate](/images/image-03.png)
+
+6. Running contract tests.
+
+Our box also comes with the file `TestSimpleStorage.js` for testing the smart contract. You can check it out in the `test` folder.
```
Comment (owans) [inline README.md]:
```suggestion
Truffle box also comes with the file `TestSimpleStorage.js` for testing the smart contract. You can check it out in the `test` folder.
```

File: README.md [hunk 3]
```diff
@@ -1,59 +1,207 @@
...
+truffle test
+```
+
+## The application
+
+Our box is a client-server: 
+- Server uses Express JS to interact with the smart contract.
+- Client uses React app.
+
+7. Running in developer mode
```
Comment (owans) [inline README.md]:
```suggestion
7. Running in development mode
```

File: README.md [hunk 4]
```diff
@@ -1,59 +1,207 @@
...
+
+In another terminal (i.e. not in the truffle develop prompt), go to the `app` directory and run the app in developer mode.
```
Comment (owans) [inline README.md]:
```suggestion
In another terminal (i.e. not in the truffle develop prompt), go to the `app` directory and run the app in development mode.
```

File: README.md [hunk 5]
```diff
@@ -1,59 +1,207 @@
...
+> Do not close the other terminal, which is running the Truffle development console, because it is our Blockchain simulator.
+> 
+> If you close it and then open it again, you need to deploy / migrate the smart contract again too!
+
+```shell
+cd app
+npm run dev
+```
+
+This command executes server and client applications and you can access in your browser:
```
Comment (owans) [inline README.md]:
```suggestion
This command executes server and client applications and you can access it in your browser:
```

File: README.md [hunk 6]
```diff
@@ -1,59 +1,207 @@
...
+- Server: [http://localhost:8080/](http://localhost:8080/)
+- Client: [http://localhost:3000/](http://localhost:3000/)
+
+> Smart contract changes must be manually recompiled and migrated!
+
+**NOTE**: This box is the starting point for the RSK tutorial [using rsk-react-express-box](https://developers.rsk.co/tutorials/truffle-boxes/rsk-react-express-box/).
+
+8. Running the production server
+
+You can chhose to run only the production server. In another terminal (i.e. not in the truffle develop prompt), go to the `app` directory and run the app in production mode.
```
Comment (owans) [inline README.md]:
```suggestion
You can choose to run only the production server. In another terminal (i.e. not in the truffle develop prompt), go to the `app` directory and run the app in production mode.
```

File: README.md [hunk 7]
```diff
@@ -1,59 +1,207 @@
...
+Take a look in the `app/dist` folder:
+
+![dist folder](/images/image-13.png)
+
+## Using RSK networks
 
 ### Setup an account & get R-BTC
 
-- Get an address using [these instructions](https://developers.rsk.co/rsk/architecture/account-based/ "Account Based RSK Addresses - RSK Developers Portal").
+- Get an address, learning how works the [account based RSK addresses](https://developers.rsk.co/rsk/architecture/account-based/ "Account based RSK addresses - RSK Developers Portal").
```
Comment (owans) [inline README.md:211]:
```suggestion
- Get an address, using the [Account based RSK addresses](https://developers.rsk.co/rsk/architecture/account-based/ "Account based RSK addresses - RSK Developers Portal") to learn how it works.
```

File: README.md [hunk 8]
```diff
@@ -62,76 +210,102 @@ First ensure you are in a new and empty directory.
...
 
-2. Find in the result the field **_minimumGasPrice_**
+This query saved the details of latest block to 
+file .minimum-gas-price-testnet.json 
+or .minimum-gas-price-mainnet.json, respectively.
 
-For more information about the **Gas** and **minimumGasPrice** please go [here](https://developers.rsk.co/rsk/rbtc/gas/ "Gas - RSK Developers Portal").
+In the `truffle-config.js`, we are reading the parameter `minimumGasPrice` in each json file.
+
+For more information about the **Gas** and **minimumGasPrice** please go to [gas page](https://developers.rsk.co/rsk/rbtc/gas/ "Gas - RSK Developers Portal").
```
Comment (owans) [inline README.md]:
```suggestion
For more information about the **Gas** and **minimumGasPrice** please go to the [gas page](https://developers.rsk.co/rsk/rbtc/gas/ "Gas - RSK Developers Portal").
```

File: README.md [hunk 9]
```diff
@@ -62,76 +210,102 @@ First ensure you are in a new and empty directory.
...
+
+```shell
+# Console for Testnet
+truffle console --network testnet
+
+# Console for Mainnet
+truffle console --network mainnet
+```
+
+3. Migrate the smart contracts. We will do it running the below commands directly in the terminal, without using the truffle console now to show to you this alternative.
```
Comment (owans) [inline README.md]:
```suggestion
3. Migrate the smart contracts. We will do it running the below commands directly in the terminal, without using the truffle console, this is to show you an alternative.
```

File: README.md [hunk 10]
```diff
@@ -62,76 +210,102 @@ First ensure you are in a new and empty directory.
...
+
+```shell
+truffle migrate
+```
+4. Update Express JS component. 
+
+The express server uses the [web3.js](https://web3js.readthedocs.io/) library to interact with the blockchain - writing code that reads and writes data from the blockchain with smart contracts.
+
+Choose which network you would like to connect the server to RSK Network
```
Comment (owans) [inline README.md]:
```suggestion
Choose which RSK network you would like to connect the server to 
```

File: README.md [hunk 11]
```diff
@@ -62,76 +210,102 @@ First ensure you are in a new and empty directory.
...
-    ```shell
-    compile
-    migrate
-    ```
+Check out the [RSK developers portal](https://developers.rsk.co/).
 
-**Then continue from step 5 of [installation steps](#installation)**
+- **Do you have questions?**
 
+Ask in [RSK chat](https://gitter.im/rsksmart/getting-started).
```
Comment (owans) [inline README.md]:
```suggestion
Ask in the [RSK Gitter Chat](https://gitter.im/rsksmart/getting-started).
```

**Review by jonathansmirnoff — APPROVED** [review]
LGTM

File: README.md (hunk 2 above)
Comment (solangegueiros) [reply README.md]: No, I am not talking about "any" Truffle box, but this specific Truffle box.

File: README.md (hunk 7 above)
Comment (solangegueiros) [reply README.md:211]: No, the page [account-based](https://developers.rsk.co/rsk/architecture/account-based/) didn't teach how to get an address. After you get an address (and I am not explain how to do it here), this page explains how the addresses work in RSK.

File: README.md (hunk 11 above)
Comment (solangegueiros) [reply README.md]: No, I wouldn't like to specify the name of the chat.

Comment (solangegueiros) [reply README.md]:
```suggestion
Ask in the [RSK chat](https://gitter.im/rsksmart/getting-started).
```

File: README.md (hunk 2 above)
Comment (solangegueiros) [reply README.md]:
```suggestion
This Truffle box also comes with the file `TestSimpleStorage.js` for testing the smart contract. You can check it out in the `test` folder.
```

**Review by alebanzas — APPROVED** [review]
LGTM. Thanks @solangegueiros !!

**Pull request merged**